src/super_gradients/training/datasets/detection_datasets/yolo_format_detection.py

```diff
@@ -0,0 +1,125 @@
+
+from super_gradients.training.datasets.data_formats import ConcatenatedTensorFormatConverter
+from super_gradients.training.datasets.data_formats.default_formats import XYXY_LABEL, LABEL_NORMALIZED_CXCYWH
+
+logger = get_logger(__name__)
+
+
+class YoloV5FormattedDetectionDataset(DetectionDataset):
+    """Base dataset to load ANY dataset that is with a similar structure to the YoloV5 dataset.
+
+    **Note**: For compatibility reasons, the dataset returns labels in Coco format (XYXY_LABEL) and NOT in YoloV5 format (LABEL_CXCYWH).
+
+    Output format: XYXY_LABEL (x, y, x, y, class_id)
+    """
+
+    def __init__(
+        self,
+        data_dir: str,
+        images_dir: str,
+        labels_dir: str,
+        classes: List[str],
+        class_ids_to_ignore: Optional[List[int]] = None,
+        *args,
+        **kwargs,
+    ):
+        """
+        :param data_dir:                Where the data is stored.
+        :param images_dir:              Name of the directory that includes all the images. Path relative to data_dir.
+        :param labels_dir:              Name of the directory that includes all the labels. Path relative to data_dir.
+        :param classes:                 List of class names.
+        :param class_ids_to_ignore:     List of class ids to ignore in the dataset. By default, doesnt ignore any class.
+        """
+        self.images_dir = images_dir
+        self.labels_dir = labels_dir
+        self.class_ids_to_ignore = class_ids_to_ignore or []
+        self.classes = classes
+
+        kwargs["target_fields"] = ["target"]
+        kwargs["output_fields"] = ["image", "target"]
+        kwargs["original_target_format"] = XYXY_LABEL  # We convert yolov5 format (LABEL_CXCYWH) to Coco format (XYXY_LABEL) when loading the annotation
+        super().__init__(data_dir=data_dir, *args, **kwargs)
+
+    @property
+    def _all_classes(self) -> List[str]:
+        return self.classes
+
+    def _setup_data_source(self) -> int:
+        """Initialize img_and_target_path_list and warn if label file is missing
+
+        :return: number of images in the dataset
+        """
+        self.images_folder = os.path.join(self.data_dir, self.images_dir)
+        self.labels_folder = os.path.join(self.data_dir, self.labels_dir)
+
+        self.images_file_names = list(sorted(os.listdir(self.images_folder)))
+        self.labels_file_names = list(sorted(os.listdir(self.labels_folder)))
+
+        image_file_base_names = set(os.path.splitext(os.path.basename(image_file_name))[0] for image_file_name in self.images_file_names)
+        label_file_base_names = set(os.path.splitext(os.path.basename(label_file_name))[0] for label_file_name in self.labels_file_names)
+        if image_file_base_names != label_file_base_names:
+            raise DatasetValidationException(f"image folder {self.images_folder} and label folder {self.labels_folder} include files that don't match")
```

Comment:
Let's not crash on this. 
A large dataset might be slightly corrupted. 
instead, log an error with the names of the missing files

Reply:
To not crash on this, we have to filter out labels/images that don't have a valid image/label file.
74a041e68abfb641b48cb0d09ff00a8041243905